src/language-js/printer-estree.js

```diff
@@ -6255,6 +6262,16 @@ function isLiteral(node) {
   );
 }
 
+function isStringPropSafeToCoerceToIdentifier(node, options) {
+  return (
+    isStringLiteral(node.key) &&
+    isIdentifierName(node.key.value) &&
+    !node.computed &&
+    options.parser !== "json" &&
+    !(options.parser === "typescript" && node.type === "ClassProperty")
```

Comment:
Anyone have the history on this? Would like to add tests for this TypeScript case.

Reply:
#4516